src/cli/integration/integration_test.go

```diff
@@ -793,13 +808,307 @@ func TestAddressGen(t *testing.T) {
 				require.Equal(t, sk.Hex(), w.Entries[0].Secret)
 			},
 		},
+		{
+			name: "addressGen --hide-secrets --mode=secrets",
+			args: []string{"addressGen", "--mode=secrets", "--hide-secrets"},
+			check: func(t *testing.T, v []byte) {
+				require.Equal(t, "secrets mode selected but hideSecrets enabled\n", string(v))
+			},
+			err: errors.New("exit status 1"),
+		},
+		{
+			name: "addressGen --mode=secrets",
+			args: []string{"addressGen", "--mode=secrets"},
+			check: func(t *testing.T, v []byte) {
+				// Confirms that only secret keys are returned
+				v = bytes.Trim(v, "\n")
+				_, err := cipher.SecKeyFromHex(string(v))
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "addressGen --mode=secrets --coin=bitcoin",
+			args: []string{"addressGen", "--mode=secrets", "--coin=bitcoin"},
+			check: func(t *testing.T, v []byte) {
+				// Confirms that only secret keys are returned
+				v = bytes.Trim(v, "\n")
+				_, err := cipher.SecKeyFromBitcoinWalletImportFormat(string(v))
+				require.NoError(t, err)
+			},
+		},
 	}
 
 	for _, tc := range tt {
 		t.Run(tc.name, func(t *testing.T) {
 			output, err := exec.Command(binaryPath, tc.args...).CombinedOutput()
+			if tc.err != nil {
+				require.Error(t, err)
+				require.Equal(t, tc.err.Error(), err.Error())
+			} else {
+				require.NoError(t, err)
+			}
+
+			tc.check(t, output)
+		})
+	}
+}
+
+func TestFiberAddressGen(t *testing.T) {
+	if !doLiveOrStable(t) {
+		return
+	}
+
+	checkFileExists := func(t *testing.T, fn string) {
+		_, err := os.Stat(fn)
+		require.NoError(t, err)
+	}
+
+	checkFileNotExists := func(t *testing.T, fn string) {
+		_, err := os.Stat(fn)
+		require.True(t, os.IsNotExist(err))
+	}
```

Comment:
I think we could move this into `testutil` as file checking is used at a of places

Reply:
added